### PR TITLE
Update cookbook to modern standards

### PR DESCRIPTION
**Pull request by wolf31o2**
Updated several items to improve the cookbook.

- Use Delivery for testing
- Use Kitchen Dokken on Ubuntu 16.04 in Travis CI
- Update metadata to include missing items/update license string
- Cleanup invalid syntax in default recipe

Signed-off-by: Chris Gianelloni <wolf31o2@gmail.com>

### CODE REVIEW

**Comment by jjasghar** [comment]
Awesome, this seems great. I'll run some local validation and get this merged ASAP. Thanks! 🤘

File: metadata.rb [hunk 1]
```diff
@@ -10,6 +10,8 @@
   respond_to?(:source_url)
 issues_url 'https://github.com/chef-partners/dcos-cookbook/issues' if
   respond_to?(:issues_url)
```
Comment (tas50) [inline metadata.rb:12]: If you require chef 12 you can nuke these respond_to and issues_url and source_url.

Comment (wolf31o2) [reply metadata.rb:12]: Lowered it to 11, since there's nothing 12+ specific here.

**Comment by jjasghar** [comment]
Thanks for this Chris!

**Pull request merged**